pkg/operator/ceph/cluster/crash/add.go

```diff
@@ -0,0 +1,188 @@
+		&handler.EnqueueRequestsFromMapFunc{
+			ToRequests: handler.ToRequestsFunc(func(obj handler.MapObject) []reconcile.Request {
+				logger.Debugf("try to reconcile deployment")
+				deployment, ok := obj.Object.(*appsv1.Deployment)
+				if !ok {
+					logger.Debugf("could not get deployment")
+					return []reconcile.Request{}
+				}
+				labels := deployment.GetLabels()
+				appName, ok := labels[k8sutil.AppAttr]
+				if !ok || appName != AppName {
+					logger.Debugf("could not find app with labels")
+					return []reconcile.Request{}
+				}
+				nodeName, ok := deployment.Spec.Template.ObjectMeta.Labels[NodeNameLabel]
+				if !ok {
+					logger.Debugf("could not get node name from label")
+					return []reconcile.Request{}
+				}
+				req := reconcile.Request{NamespacedName: types.NamespacedName{Name: nodeName}}
+				logger.Debugf("try to reconcile the ceph-crash deployment")
+				return []reconcile.Request{req}
+			}),
+		},
+	)
+	if err != nil {
+		return fmt.Errorf("failed to watch for changes on the ceph-crash deployment. %+v", err)
+	}
+
+	// Watch for changes to the ceph pod nodename and enqueue their nodes
+	logger.Debugf("watch for changes to the ceph pod nodename and enqueue their nodes")
+	err = c.Watch(
+		&source.Kind{Type: &corev1.Pod{}},
+		&handler.EnqueueRequestsFromMapFunc{
+			ToRequests: handler.ToRequestsFunc(func(obj handler.MapObject) []reconcile.Request {
+				pod, ok := obj.Object.(*corev1.Pod)
+				if !ok {
+					logger.Debugf("could not get pod")
+					return []reconcile.Request{}
+				}
+				nodeName := pod.Spec.NodeName
+				if nodeName == "" {
+					logger.Debugf("could not get node name from the pod spec")
+					return []reconcile.Request{}
+				}
+				if isCephPod(pod.Labels, pod.Name) {
+					req := reconcile.Request{NamespacedName: types.NamespacedName{Name: nodeName}}
+					logger.Debugf("req is %+v", req)
+					return []reconcile.Request{req}
+				}
+				logger.Debugf("%q is not a ceph pod", pod.Name)
+				return []reconcile.Request{}
+			}),
+		},
+		// only enqueue the update event if the pod moved nodes
+		predicate.Funcs{
+			UpdateFunc: func(event event.UpdateEvent) bool {
+				oldPod, ok := event.ObjectOld.(*corev1.Pod)
+				if !ok {
+					logger.Debugf("could not get old pod")
+					return false
+				}
+				newPod, ok := event.ObjectNew.(*corev1.Pod)
+				if !ok {
+					logger.Debugf("could not get new pod")
+					return false
+				}
+				// only enqueue if the nodename has changed
+				if oldPod.Spec.NodeName == newPod.Spec.NodeName {
+					logger.Debugf("old pod and new pod are the same")
+					return false
+				}
+				logger.Debugf("enqueue update event since new pod name is %q and old pod name is %q", newPod.Spec.NodeName, oldPod.Spec.NodeName)
+				return true
+			},
+		},
+	)
+	if err != nil {
+		return fmt.Errorf("failed to watch for changes on the ceph pod nodename and enqueue their nodes. %+v", err)
+	}
+
+	// watch for change in the ceph cluster and reconcile the owner
+	err = c.Watch(
```

Comment:
This watch won't do anything as we are not setting an owner reference on the cephcluster for the node. My bad!

Reply:
Actually I don't think it was there at first, it's a leftover from my attempts, good catch!